test/pre-post-extension.js

```diff
@@ -0,0 +1,56 @@
+import chai, {expect} from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import Stipulate from '../src';
+
+chai.use(chaiAsPromised);
+
+class ExtStip extends Stipulate {
```

Comment:
You're testing the EMCA 6 specification for class extension, which shouldn't be tested. 

In the index test file, you should be asserting that the hooks are called before and after the fetch--basically, that the hooks are called when they're supposed to be called.

Extension is clearly defined and an expected behavior. If the hooks are called in `send`, then it follows that they'll be called in an extended class too.